### ISSUE TITLE
Queue/Metrics utilization dashboard for usage overview

### ISSUE THREAD

**Issue by pat-s** [woodpecker-ci/woodpecker#3105]
### Clear and concise description of the problem

Currently, tracking usage pattern is hard for admins as there is no easy way to get queue statistics over time. The API endpoint `queue/info` provides some basic stats but one would need to query it every few seconds/minute and then post-process the data manually into a dashboard.

Yes, there is a prometheus endpoint at `/metrics` and some user-provided dasbhoard (e.g. https://github.com/Janik-Haag/woodpecker-grafana-dashboard) but having a built-in view would be really great.

### Suggested solution

Add a simple dashboard to `admin#queue` which shows queue utilization over time. 

Bonus: group the stats by agent (e.g. to infer how many builds for a given architecture were requested)

https://github.com/arl/statsviz might be a library for this.

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by usiegj00** [comment]
I would like to implement this, and I have a design that avoids the part I expect you to object to: storing history in the database.

**Context.** I run a Woodpecker instance with 18 agents and 16 Linux slots. Today a backlog of 35 pull-request gates (10 legs each) built up, and I could not answer the four questions an admin actually has: how many runners are live, how deep is the backlog, how fast is it draining, and when will it clear. `/metrics` can answer them only if you already know the query language, and `queue/info` is a point sample with no history.

**Proposal A — no new storage (preferred).** Everything except one number is already in the database:

- **Throughput**: `SELECT count(*) FROM workflows WHERE finished BETWEEN ? AND ?`, bucketed.
- **Depth over time**: workflows whose pipeline was created before T and had not started by T. This is an approximation (it counts workflows blocked on dependencies as queued), but it tracks the shape of the backlog well.
- **Now**: the existing `queue/info` counters.
- **Runners**: the `agents` table.

Measured on my production instance (Postgres, ~9.6k steps, ~2.5k workflows):

| query | time |
|---|---|
| current backlog, joined to live pipelines | 0.98 ms |
| 7-day hourly throughput | 6.2 ms |
| 7-day hourly depth reconstruction | 27.8 ms |

It produced exactly the view I wanted: depth 28 → 157 → 592 → 714 across the morning, running pinned at 25 (the slot ceiling), throughput ~150 workflows/hour, so "clears in about 4.8 h".

This needs **one index on `workflows(finished)`** and no new table, no sampler, no retention policy, and no write on the hot path. It also cannot conflict with the queue refactor in #6317, because it reads finished rows only.

**Proposal B — recorded series, only if you want exact numbers.** If the approximation is not good enough (it cannot show waiting-on-deps, real queue wait time, or agent capacity over time), then a small table written by a **new decorator in the queue chain** (`server/queue/queue.go`, beside `WithTaskStore`), not by touching `fifo.go`. Off unless `WOODPECKER_QUEUE_STATS=true`, so nothing is written by default; rows bounded by a retention window; no migration needed since the bean sync creates the table. I would only build this if you prefer it.

**UI.** The admin queue page already exists, so this is a panel on it rather than a new page. You have no charting dependency, and I would not add one: hand-written SVG, in keeping with the existing stacked-bar stats component.

**Questions before I write code**
1. Is Proposal A (query-only, one index) acceptable, or do you consider even that out of scope for the server?
2. If you would take B instead, is a queue decorator the right seam given #6317?
3. Is a hand-written SVG chart acceptable, or would you rather have no chart and only current numbers with a drain estimate?

**Unrelated defect I noticed while reading the code.** `AdminQueueStats.vue` shows `completed_count` as its headline number and `web/src/lib/api/types/queue.ts` declares it, but the server's `QueueInfo` model has only `worker_count`, `pending_count`, `waiting_on_deps_count` and `running_count`. I confirmed against a live 3.18.1 server: the field is absent from the response, so that number renders undefined. Happy to send a separate small PR for it — tell me whether you want the field added server-side or the UI changed.

**Comment by qwerty287** [comment]
I agree storing the full history over time sounds a bit too much, so I like your proposal A. But I'm not sure whether this works fine, because for example we are using the autoscaler and thus the number of agents can be very different. Then this would give really bad results. And I think there are probably quite some people use the autoscaler and they would be affected by this.

> Unrelated defect I noticed while reading the code. AdminQueueStats.vue shows completed_count as its headline number and web/src/lib/api/types/queue.ts declares it, but the server's QueueInfo model has only worker_count, pending_count, waiting_on_deps_count and running_count. I confirmed against a live 3.18.1 server: the field is absent from the response, so that number renders undefined. Happy to send a separate small PR for it — tell me whether you want the field added server-side or the UI changed.

Would just remove it from the UI.

**Comment by usiegj00** [comment]
This is what I'm playing with using the data that's already in the db. I'll look into autoscaling and how it would be affected.
<img width="1974" height="852" alt="Image" src="https://github.com/user-attachments/assets/8e358e28-42c7-4f23-99b7-51b83084934d" />
